Plutarch/Builtin.hs

```diff
@@ -201,12 +201,7 @@ instance PIsData a => PIsData (PBuiltinList (PAsData a)) where
   pfromData x = punsafeCoerce $ pasList # pforgetData x
   pdata x = punsafeBuiltin PLC.ListData # x
 
-instance
-  ( PIsData k
```

Comment:
Not entirely sure if removing these constraints is a good idea. They are certainly redundant in this case. But it didn't make any sense to do `pdata` on a `PBuiltinMap k v` where `k` and `v` don't have `PIsData` impls.

Then again, if they don't have `PIsData` impls, you wouldn't be able to build the `PAsData k` and `PAsData v` required anyway. Which reminds me, what's the recommended way to build `PBuiltinMap`?

Reply:
A constant? MkPairData and MkCons? Something like that. There are situations where `PIsData k` is not fulfillable, even though it's in a `PAsData`, e.g. `PDataList`. In addition, reducing the constraints here also means reducing the number of constraints downstream.